app/config_test.go

```diff
@@ -110,23 +113,38 @@ func TestDBConfig_Success(t *testing.T) {
 	globalConfig := viper.New()
 	_ = os.Setenv("", "myself")
 	globalConfig.Set("database.collation", "stuff")
-	globalConfig.Set("database.TLSConfig", "v88")
-	// Still buggy, for unmarshaled config, the config needs to be set first (by config file
-	// or manually) to allow setting it through env
-	_ = os.Setenv("ALGOREA_DATABASE__TLSCONFIG", "v99")
+	_ = os.Setenv("ALGOREA_DATABASE__TLSCONFIG", "v99") // env var which was not defined before
 	defer func() { _ = os.Unsetenv("ALGOREA_DATABASE__TLSCONFIG") }()
 	dbConfig, err := DBConfig(globalConfig)
 	assert.NoError(err)
 	assert.Equal("stuff", dbConfig.Collation)
 	assert.Equal("v99", dbConfig.TLSConfig)
 }
 
-func TestDBConfig_Error(t *testing.T) {
+func TestDBConfig_UnmarshallingError(t *testing.T) {
+	// don't know if it is really possible to get this error
 	assert := assertlib.New(t)
 	globalConfig := viper.New()
-	globalConfig.Set("database.Timeout", "invalid")
+	monkey.PatchInstanceMethod(reflect.TypeOf(globalConfig), "Unmarshal",
+		func(_ *viper.Viper, _ interface{}, _ ...viper.DecoderConfigOption) error {
+			return fmt.Errorf("unmarshalling error")
+		},
+	)
+	defer monkey.UnpatchAll()
+	_, err := DBConfig(globalConfig)
+	assert.EqualError(err, "unmarshalling error")
+}
+
+func TestDBConfig_StructToMapError(t *testing.T) {
+	// unexpected error, must monkey patch it
+	assert := assertlib.New(t)
+	globalConfig := viper.New()
+	monkey.Patch(mapstructure.Decode, func(_ interface{}, _ interface{}) error {
+		return fmt.Errorf("struct2map error")
```

Comment:
errors.New(...)?

Reply:
Is it really better? Both are used in the code. I can change it.

Reply:
I see that many other places in the code are doing this way, so I'll keep it here as well (it is a test anyway).